include/fmt/chrono.h

```diff
@@ -430,6 +430,45 @@ auto write(OutputIt out, const std::tm& time, const std::locale& loc,
   return write_encoded_tm_str(out, string_view(buf.data(), buf.size()), loc);
 }
 
+#if FMT_SAFE_DURATION_CAST
+// throwing version of safe_duration_cast
+// only available for integer<->integer or float<->float casts
+template <typename To, typename FromRep, typename FromPeriod,
+          FMT_ENABLE_IF((std::is_integral<FromRep>::value &&
+                         std::is_integral<typename To::rep>::value) ||
+                        (std::is_floating_point<FromRep>::value &&
+                         std::is_floating_point<typename To::rep>::value))>
```

Comment:
Let's add a trait for this and reuse here and below (negated).

Reply:
Done